File: frontend/src/app/models/kategorija.ts
import { Atribut } from "./atribut";

export class Kategorija {

    kategorijaId! : number;
    nazivKategorije : string;
    atribut! : Atribut;

    constructor(nazivKategorije : string){
        this.nazivKategorije = nazivKategorije;
    }
}

File: backend/src/main/java/com/ip/fitnessApp/model/Kategorija.java
package com.ip.fitnessApp.model;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Data
@EqualsAndHashCode(callSuper = false)
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Kategorija {
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Id
    @Column(name = "kategorija_id")
    private int kategorijaId;
    @Basic
    @Column(name = "naziv_kategorije")
    private String nazivKategorije;

    @ManyToOne
    @JoinColumn(name = "atribut_id")
    private Atribut atribut;

}
